### PR TITLE
dnsapi: add deSEC.io api support

### PR DESCRIPTION
**Pull request by sotux**

### CODE REVIEW

**Comment by Neilpang** [comment]
read this guide first:
https://github.com/Neilpang/acme.sh/wiki/DNS-API-Dev-Guide

**Comment by sotux** [comment]
Hi @Neilpang , I've updated the script. Please review.
If it still has some mistakes, please let me know. Thanks.

**Comment by Neilpang** [comment]
please read the items in the guide one by one:
https://github.com/Neilpang/acme.sh/wiki/DNS-API-Dev-Guide

**Comment by sotux** [comment]
Hi @Neilpang, I've update the scripts and add an entry in README.md for deSEC.io api.
This version could support wildcard cert.

File: dnsapi/dns_desec.sh [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+_desec_rest() {
+  m="$1"
+  ep="$2"
+  data="$3"
+
+  export _H1="Authorization: Token $DEDYN_TOKEN"
+  export _H2="Accept: application/json"
+  export _H3="Content-Type: application/json"
+  # clear headers from previous request to avoid getting wrong http code on timeouts
+  :>"$HTTP_HEADER"
```
Comment (Neilpang) [inline dnsapi/dns_desec.sh]: remove this line,  you dont need to clear the http header file.

**Comment by Neilpang** [comment]
please move your usage here:
https://github.com/Neilpang/acme.sh/wiki/dnsapi

please resolve the conflicts

**Comment by sotux** [comment]
> please move your usage here:
> https://github.com/Neilpang/acme.sh/wiki/dnsapi
> 
> please resolve the conflicts

Have rebased & edited the wiki

**Pull request merged**